internal/pkg/crawl/link_header.go

```diff
@@ -0,0 +1,52 @@
+package crawl
+
+import (
+	"regexp"
+	"strings"
+)
+
+// Represents a Link struct, containing a URL to which it links, and a Rel to define the relation
+type Link struct {
+	URL string
+	Rel string
+}
+
+// Parse parses a raw Link header in the form:
+//
+//	<url1>; rel="what", <url2>; rel="any"; another="yes", <url3>; rel="thing"
+//
+// returning a slice of Link structs
+// Each of these are separated by a `, ` and the in turn by a `; `, with the first always being the url, and the remaining the key-val pairs
+// See: https://simon-frey.com/blog/link-header/, https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Link
+func Parse(link string) []Link {
+	var links []Link
+
+	urlRegex := regexp.MustCompile(`^<(\S+)>$`)
+	// var linkRegex = regexp.MustCompile(`^<(\S+)>(; (\S+)=(\S+))*$`)
+
+	for _, link := range strings.Split(link, ", ") {
+		parts := strings.Split(link, "; ")
+		match := urlRegex.FindStringSubmatch(parts[0])
+		url := match[1]
```

Comment:
If there is no match and the value of match is `nil`, then the program will halt due to trying to access an index on a `nil` value. The function needs to be written to handle malformed inputs.

Reply:
There should also be a test added to demonstrate the handling of malformed input

Reply:
Test Added